dbms/src/Storages/Page/V3/PageStorageImpl.cpp

```diff
@@ -47,6 +47,8 @@ void PageStorageImpl::restore()
 {
     // TODO: clean up blobstore.
     // TODO: Speedup restoring
+    blob_store.registerPaths();
```

Comment:
If there are BlobFile 1, 2, 3 on disks, but only BlobFile 2, 3 have valid data. Then calling `registerPaths` will register BlobFile 1 on `delegator`, but actually, BlobFile 1 on disk will be removed in `BlobStats::restore`. But it won't be removed from the `delegator`.
https://github.com/pingcap/tiflash/blob/924ea6d67f8931a36523cb1db6cbcbbe4161c740/dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp#L96

Maybe we should call `delegator->addPageFileUsedSize` in `BlobStats::restore`?

Reply:
Can't call `addPageFileUsedSize` in `BlobStats::restore`. `addPageFileUsedSize`  or `registerPaths()` can't be call after `restoreByEntry`. Becasue we will find path in `restoreByEntry`.

Added `removePageFile` in `BlobStats::restore()`.

Reply:
changed